module/scripts/rdf-schema-alignment.js

```diff
@@ -1,336 +1,363 @@
+      }
+      if (overwrite) {
+        self._nodeUIs = [];
+        self._renderBody(body);
+        DialogSystem.dismissUntil(self._level);
+      } else {
+        return;
+      }
+    }, function(rdfSkeleton) {
+      $.getJSON("command/core/get-operations?" + $.param({
+        project: theProject.id
+      }), function(json) {
+        var operations_old = json.entries.map(function(val) {
+          return val['operation'];
+        });
         
-        Refine.postProcess(
-                "rdf-extension",
-                "save-rdf-schema",
-                {},
-                { schema: JSON.stringify(schema) },
-                {},
-                {   
-                        onDone: function() {
-                        theProject.overlayModels.rdfSchema = schema;
-                        }
-                }
-        );
-   });
-    
-    elmts._share_skeleton.click(function(e){
-    	e.preventDefault();
-    	var schema = self.getJSON();
-    	rdfSkeletonMetadataDialog = new RdfSkeletonMetadataDialog(function(){
-            var schema = self.getJSON();
-            var title = $("input#title").val();
-            var fileType = $("input#fileType").val();
-            
-            if (fileType != null){
-                $.post("command/rdf-extension/save-rdf-skeleton",
-                        {model: JSON.stringify(self.getJSON()), 
-                    projectId: theProject.id, fileType: fileType, title: title},
-                    function(data){
-                        alert('RDF skeleton saved sucessfully');
-                    }
-                );
+        if (operations_old.length == 1) {
+          operations_old = [];
+        } else {
+          operations_old = operations_old.slice(-5);
+        }
+        var operations_new = JSON.parse(rdfSkeleton['skeleton']).operations;
+        Refine.postCoreProcess("apply-operations", {}, {
+          operations: JSON.stringify(operations_old.concat(operations_new))
+        }, {
+          everythingChanged: true
+        }, {
+          onDone: function(o) {
+            if (o.code == "pending") {
+              Refine.update({
+                everythingChanged: true
+              });
             }
-            
-            
-            Refine.postProcess(
-                    "rdf-extension",
-                    "save-rdf-schema",
-                    {},
-                    { schema: JSON.stringify(schema) },
-                    {},
-                    {   
-                        onDone: function() {
-                           theProject.overlayModels.rdfSchema = schema;
-                        }
-                    }
-                );
-    	});
-    });
+          }
+        });
+      });
+      var prefixes = [];
+      var new_prefixes = JSON.parse(rdfSkeleton['skeleton']).prefixes;
+      if (self._schema.prefixes != undefined) {
+        prefixes = prefixes.concat(self._schema.prefixes).concat(new_prefixes);
```

Comment:
Not too familiar with the setup, so just some questions:
- `rdfSkeleton` is the current model?
- `self._schema` is the new model?

@rajaram5 what do you think, should we have this included for the demo?

Reply:
RDf skeleton is from the json blob and self._schema is the instance in the fairifier

Reply:
@Shamanou didn't we decide to override the skeleton?

Reply:
@kburger for the demo it is enough if we can do skeleton overwriting

Reply:
@rajaram5 yeah I meant whether or not to include this change, not necessarily to demo it, but to prevent potential bugs from messing up the demo.

Reply:
@Shamanou after discussion with @rajaram5, we think it would be more robust if you could do more checks on the prefixes that are loaded from the json blob. Check for existence, whether it is empty, etc.

Reply:
ok will do

Reply:
line 127 checks for existence, new commit will check for empty prefix list